service/model/project.go

```diff
@@ -0,0 +1,91 @@
+package model
+
+import (
+	"context"
+	"time"
+
+	svcerr "release-manager/service/errors"
+
+	"github.com/google/uuid"
+)
+
+type ProjectRepository interface {
+	Create(ctx context.Context, p Project) (Project, error)
+	Read(ctx context.Context, id uuid.UUID) (Project, error)
+	ReadAll(ctx context.Context) ([]Project, error)
+	Update(ctx context.Context, p Project) (Project, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+}
+
+type Project struct {
+	ID                 uuid.UUID
+	Name               string
+	Description        string
+	ReleaseAlertConfig ReleaseAlertConfig
+	SlackConfig        SlackConfig
+	CreatedAt          time.Time
+	UpdatedAt          time.Time
+}
+
+type ProjectUpdate struct {
+	Name               *string
+	Description        *string
+	ReleaseAlertConfig *ReleaseAlertConfig
+	SlackConfig        *SlackConfig
+}
+
+type SlackConfig struct {
+	ChannelID string
+}
+
+func NewProject(name, description string, rlsCfg ReleaseAlertConfig, slackCfg SlackConfig) (Project, error) {
+	p := Project{
+		ID:                 uuid.New(),
+		Name:               name,
+		Description:        description,
+		ReleaseAlertConfig: rlsCfg,
+		SlackConfig:        slackCfg,
+		CreatedAt:          time.Now(),
+		UpdatedAt:          time.Now(),
+	}
+
+	if err := p.Validate(); err != nil {
+		return Project{}, err
+	}
+
+	return p, nil
+}
+
+func (p *Project) Update(name, description *string, rlsCfg *ReleaseAlertConfig, slackCfg *SlackConfig) error {
+	if name != nil {
+		p.Name = *name
+	}
+	if description != nil {
+		p.Description = *description
+	}
+	if rlsCfg != nil {
+		p.ReleaseAlertConfig = *rlsCfg
+	}
+	if slackCfg != nil {
+		p.SlackConfig = *slackCfg
+	}
+	p.UpdatedAt = time.Now()
+
+	if err := p.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Project) Validate() error {
```

Comment:
Validate function enforces all business rules and is called when creating or updating project object. 

I prefer to not use any validator package in service layer.